data/src/test/java/org/apache/iceberg/data/GenericAppenderHelper.java

```diff
@@ -45,34 +44,19 @@ public class GenericAppenderHelper {
 
   private final Table table;
   private final FileFormat fileFormat;
-  private final TemporaryFolder tmp;
-  private final Path temp;
+  private final Path tempDir;
   private final Configuration conf;
 
-  @Deprecated
   public GenericAppenderHelper(
-      Table table, FileFormat fileFormat, TemporaryFolder tmp, Configuration conf) {
+      Table table, FileFormat fileFormat, Path tempDir, Configuration conf) {
```

Comment:
I don't think you can make changes in this file at this point, because there are still places across the codebase that use this and pass a `TemporaryFolder`. That being said, please revert the changes to this file. We can only remove this constructor once all the other places (most likely Flink) are converted to JUnit5 (which isn't the scope of this PR)

Reply:
My original idea was to remove the Junit 4 dependencies completely from the `data` module and while doing so follow the trails to other modules when necessary (mostly `flink`). I understand your preference is rather to remove as much Junit 4 stuff from `data` as possible while minimizing the impact on other modules. So, I added back the constructor to `GenericAppenderHelper` that accepts `TemporaryFolder` (and marked it as `@Deprecated`) and I reverted the related changes in `flink`. I believe the `TemporaryFolder` argument in this constructor is the only one remaining dependency to Junit 4 in the `data` module. Pls LMK if this is OK.

Having said that, I haven't reverted all my changes in this file because the other constructor seems fine to have been removed as the `@Deprecated` `TestHelper` constructor in the `mr` module isn't used anymore and could be removed. But pls double check and LMK.